File: presentation/src/main/java/info/firozansari/presentation/AndroidApplication.java
package info.firozansari.presentation;

import android.app.Application;
import info.firozansari.presentation.internal.di.components.ApplicationComponent;
import info.firozansari.presentation.internal.di.components.DaggerApplicationComponent;
import info.firozansari.presentation.internal.di.modules.ApplicationModule;


/**
 * Android Main Application
 */
public class AndroidApplication extends Application {

  private ApplicationComponent applicationComponent;

  @Override public void onCreate() {
    super.onCreate();
    this.initializeInjector();
  }

  private void initializeInjector() {
    this.applicationComponent = DaggerApplicationComponent.builder()
        .applicationModule(new ApplicationModule(this))
        .build();
  }

  public ApplicationComponent getApplicationComponent() {
    return this.applicationComponent;
  }

}

File: presentation/src/main/java/info/firozansari/presentation/AndroidApplication.kt
package info.firozansari.presentation

import android.app.Application
import info.firozansari.presentation.internal.di.components.DaggerApplicationComponent

/**
 * Android Main Application
 */
class AndroidApplication : Application() {
    private var applicationComponent: ApplicationComponent? = null
    override fun onCreate() {
        super.onCreate()
        initializeInjector()
    }

    private fun initializeInjector() {
        applicationComponent = DaggerApplicationComponent.builder()
            .applicationModule(ApplicationModule(this))
            .build()
    }

    fun getApplicationComponent(): ApplicationComponent? {
        return applicationComponent
    }
}